File: Chapter 04/JS/4.4 Anti-Patterns/After-RichDomain_ThinController/domain_models/customer.js
// Chapter 04/JavaScript/4.4 Anti-Patterns/After-RichDomain/domain_models/customer.js

/**
 * ARCHITECTURE NOTE: Not every domain model needs complex behavior. 
 * Because the core business rules for this bounded context revolve 
 * around the Order, this Customer class can remain a simple data 
 * entity holding state.
 */
class Customer {
    constructor() {
        this.id = 0;
        this.type = ""; // e.g., "Gold"
        this.email = "";
    }
}

module.exports = Customer;

File: Chapter 04/Python/section_4_4_anti_patterns/after_rich_domain_thin_controller/domain_models/customer.py
# Chapter 04/Python/4.4 Anti-Patterns/After-RichDomain/domain_models/customer.py

class Customer:
    """
    ARCHITECTURE NOTE: Not every domain model needs complex behavior. 
    Because the core business rules for this bounded context revolve 
    around the Order, this Customer class can remain a simple data 
    entity holding state.
    """
    def __init__(self):
        self.id = 0
        self.type = ""  # e.g., "Gold"
        self.email = ""